### PR TITLE
Only import PNG files if the target has saveAsPNG turned on

### PR DESCRIPTION
**Pull request by riknoll**
I was copying and pasting a block in the blocks editor and it caused the entire app to freeze. I guess I had a PNG on my clipboard (not sure why copying a block didn't overwrite that) because it was hanging in the decode function. I don't know why it hung, but we shouldn't try to load PNG's in non-PNG supporting editors anyhow.

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -1911,7 +1911,7 @@ function isProjectFile(filename: string): boolean {
 }
 
 function isPNGFile(filename: string): boolean {
-    return /\.png$/i.test(filename)
+    return pxt.appTarget && pxt.appTarget.compile && pxt.appTarget.compile.saveAsPNG && /\.png$/i.test(filename);
```
Comment (mmoskal) [inline webapp/src/app.tsx]: I think you just need `return pxt.appTarget.compile.saveAsPNG && /\.png$/i.test(filename);`

Comment (riknoll) [reply webapp/src/app.tsx]: You definitely need the compile check, but yeah you're right about checking for appTarget. I was a tad overzealous and forgot this is in the webapp 😄

Comment (mmoskal) [reply webapp/src/app.tsx]: No you don't. https://github.com/Microsoft/pxt/blob/master/pxtlib/main.ts#L20

Comment (riknoll) [reply webapp/src/app.tsx]: I stand corrected! We definitely check it all over the place though.

Comment (mmoskal) [reply webapp/src/app.tsx]: I know... wasting precious cycles...

**Pull request merged**